kore/src/Kore/Parser/Parser.y

```diff
@@ -0,0 +1,452 @@
+                    | forall '{' Sort '}' '(' ElementVariable ',' Pattern ')'
+                      { embedParsedPattern $ ForallF Forall{forallSort = $3, forallVariable = $6, forallChild = $8} }
+                    | mu '{' '}' '(' SetVariable ',' Pattern ')'
+                      { embedParsedPattern $ MuF Mu{muVariable = $5, muChild = $7} }
+                    | nu '{' '}' '(' SetVariable ',' Pattern ')'
+                      { embedParsedPattern $ NuF Nu{nuVariable = $5, nuChild = $7} }
+                    | ceil '{' Sort ',' Sort '}' '(' Pattern ')'
+                      { embedParsedPattern $ CeilF Ceil{ceilOperandSort = $3, ceilResultSort = $5, ceilChild = $8} }
+                    | floor '{' Sort ',' Sort '}' '(' Pattern ')'
+                      { embedParsedPattern $ FloorF Floor{floorOperandSort = $3, floorResultSort = $5, floorChild = $8} }
+                    | equals '{' Sort ',' Sort '}' '(' Pattern ',' Pattern ')'
+                      { embedParsedPattern $ EqualsF Equals{equalsOperandSort = $3, equalsResultSort = $5, equalsFirst = $8, equalsSecond = $10} }
+                    | in '{' Sort ',' Sort '}' '(' Pattern ',' Pattern ')'
+                      { embedParsedPattern $ InF In{inOperandSort = $3, inResultSort = $5, inContainedChild = $8, inContainingChild = $10} }
+                    | next '{' Sort '}' '(' Pattern ')'
+                      { embedParsedPattern $ NextF Next{nextSort = $3, nextChild = $6} }
+                    | rewrites '{' Sort '}' '(' Pattern ',' Pattern ')'
+                      { embedParsedPattern $ RewritesF Rewrites{rewritesSort = $3, rewritesFirst = $6, rewritesSecond = $8} }
+                    | dv '{' Sort '}' '(' StringLiteralPattern ')'
+                      { embedParsedPattern $ DomainValueF DomainValue{domainValueSort = $3, domainValueChild = $6} }
+                    | Id SortsBrace Patterns
+                      { embedParsedPattern $ ApplicationF Application{applicationSymbolOrAlias = SymbolOrAlias{symbolOrAliasConstructor = $1, symbolOrAliasParams = $2}, applicationChildren = $3} }
+
+Application :: {Application SymbolOrAlias (SomeVariable VariableName)}
+	     : Id SortsBrace SomeVariables { Application{applicationSymbolOrAlias = SymbolOrAlias{symbolOrAliasConstructor = $1, symbolOrAliasParams = $2}, applicationChildren = $3} }
+
+SomeVariables :: {[SomeVariable VariableName]}
+	       : '(' SomeVariableList ')' { reverse $2 }
+               | '(' ')' { [] }
+
+SomeVariableList :: {[SomeVariable VariableName]}
+		  : SomeVariable { [$1] }
+                  | SomeVariableList ',' SomeVariable { $3 : $1 }
+
+Patterns :: {[ParsedPattern]}
+	  : NePatterns { $1 }
+          | '(' ')' { [] }
+
+NePatterns :: {[ParsedPattern]}
+	    : '(' PatternList ')' { reverse $2 }
+
+PatternList :: {[ParsedPattern]}
+             : Pattern { [$1] }
+	     | PatternList ',' Pattern { $3 : $1 }
+
+{
+
```

Comment:
```suggestion
type Parser a = FilePath -> Text -> Either String a
```
Export this type alias and use it elsewhere as well (including tests). It can also clean up the following functions, see below.